### PR TITLE
[Intents] Update bindings to Xcode 12.5 from Beta 1 to Beta 3

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: src/intents.cs [hunk 1]
```diff
@@ -1767,6 +1767,8 @@ public enum INMediaDestinationType : long {
 	public enum INMediaReference : long {
 		Unknown = 0,
 		CurrentlyPlaying,
+		[Watch (7,3), TV (14,5), iOS (14,5)]
```
Comment (spouliot) [inline src/intents.cs:1770]: ^ @rachelkang pointed out to me that the SDK shipping with Xcode 12.5 is for watchOS 7.4 (not 7.3)

even if Xcode 12.4 shipped the SDK for 7.2 😢 

I need to check my own PR too... I likely made that error too

**Review by spouliot — CHANGES_REQUESTED** [review]

File: src/intents.cs (hunk 1 above)
Comment (dalexsoto) [reply src/intents.cs]: @rachelkang / @spouliot Nice catch!! Fixed, thanks!

File: src/intents.cs [hunk 2]
```diff
@@ -8939,20 +8941,40 @@ interface INCallDestinationTypeResolutionResult {
 	[DisableDefaultCtor]
 	interface INCallRecord : NSCopying, NSSecureCoding {
 
+		[Watch (7,4), Mac (11,3), iOS (14,5)]
+		[Export ("initWithIdentifier:dateCreated:callRecordType:callCapability:callDuration:unseen:participants:numberOfCalls:isCallerIdBlocked:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string identifier, [NullAllowed] NSDate dateCreated, INCallRecordType callRecordType, INCallCapability callCapability, [NullAllowed] [BindAs (typeof (double?))] NSNumber callDuration, [NullAllowed] [BindAs (typeof (bool?))] NSNumber unseen, [NullAllowed] INPerson[] participants, [NullAllowed] [BindAs (typeof (int?))] NSNumber numberOfCalls, [NullAllowed] [BindAs (typeof (bool?))] NSNumber isCallerIdBlocked);
+
+		[Deprecated (PlatformName.iOS, 14, 5, message: "Use '.ctor (string, NSDate, INCallRecordType, INCallCapability, double?, bool?, int?)' instead.")]
+		[Deprecated (PlatformName.WatchOS, 7, 3, message: "Use '.ctor (string, NSDate, INCallRecordType, INCallCapability, double?, bool?, int?)' instead.")]
```
Comment (spouliot) [inline src/intents.cs:8950]: one more ;-)

File: src/intents.cs [hunk 3]
```diff
@@ -8939,20 +8941,40 @@ interface INCallDestinationTypeResolutionResult {
...
-		[DesignatedInitializer]
 		IntPtr Constructor (string identifier, [NullAllowed] NSDate dateCreated, [NullAllowed] INPerson caller, INCallRecordType callRecordType, INCallCapability callCapability, [NullAllowed] [BindAs (typeof (double?))] NSNumber callDuration, [NullAllowed] [BindAs (typeof (bool?))] NSNumber unseen, [NullAllowed] [BindAs (typeof (int?))] NSNumber numberOfCalls);
 
+		[Watch (7,4), Mac (11,3), iOS (14,5)]
+		[Export ("initWithIdentifier:dateCreated:callRecordType:callCapability:callDuration:unseen:numberOfCalls:")]
+		IntPtr Constructor (string identifier, [NullAllowed] NSDate dateCreated, INCallRecordType callRecordType, INCallCapability callCapability, [NullAllowed] [BindAs (typeof (double?))] NSNumber callDuration, [NullAllowed] [BindAs (typeof (bool?))] NSNumber unseen, [NullAllowed] [BindAs (typeof (int?))] NSNumber numberOfCalls);
+
+		[Deprecated (PlatformName.iOS, 14, 5, message: "Use '.ctor (string, NSDate, INCallRecordType, INCallCapability, double?, bool?)' instead.")]
+		[Deprecated (PlatformName.WatchOS, 7, 3, message: "Use '.ctor (string, NSDate, INCallRecordType, INCallCapability, double?, bool?)' instead.")]
```
Comment (spouliot) [inline src/intents.cs:8961]: two :)

File: src/intents.cs [hunk 4]
```diff
@@ -8939,20 +8941,40 @@ interface INCallDestinationTypeResolutionResult {
...
+		IntPtr Constructor (string identifier, [NullAllowed] NSDate dateCreated, INCallRecordType callRecordType, INCallCapability callCapability, [NullAllowed] [BindAs (typeof (double?))] NSNumber callDuration, [NullAllowed] [BindAs (typeof (bool?))] NSNumber unseen);
+
 		[Export ("identifier")]
 		string Identifier { get; }
 
 		[NullAllowed, Export ("dateCreated", ArgumentSemantic.Copy)]
 		NSDate DateCreated { get; }
 
+		[Deprecated (PlatformName.iOS, 14, 5)]
+		[Deprecated (PlatformName.WatchOS, 7, 3)]
```
Comment (spouliot) [inline src/intents.cs:8977]: 3!

**Review by spouliot — CHANGES_REQUESTED** [review]

**Comment by dalexsoto** [comment]
@spouliot SO MANY ATTRIBUTES! 😂 Fixed!! Thanks!!!

**Review by spouliot — APPROVED** [review]
~3~

**Review by rolfbjarne — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Pull request merged**